spec/rspec/core/reporter_spec.rb

```diff
@@ -160,6 +160,18 @@ module RSpec::Core
       end
     end
 
+    describe "#exit_early" do
+      it "returns the passed exit code" do
+        expect(reporter.exit_early(42)).to eq(42)
+      end
+
+      it "reports zero examples" do
+        allow(reporter).to receive(:report)
+        reporter.exit_early(42)
+        expect(reporter).to have_received(:report).with(0)
```

Comment:
Nearly there! This should assert on the output, take the result of "start" and "finish" as an example :)

Reply:
I need you to help me out here to write exactly what you want me to write. I looked at those specs and as I understand them, they set some expectations on method calls the formatter should receive. I guess my strategy would be to double the formatter, register it as a listener, run the spec, see which unpermitted methods it receives and then stub all of them? Is that what you want me to do?

Reply:
I can see why those specs work like that because those methods directly notify the formatter. But in this case, since all that is delegated to the `#report` method... Isn't the stub I'm using now more appropriate? In case you disagree and I should still stub whatever calls the formatter receives, could you also tell me how I should word the example to match the new implementation? Thanks!

Reply:
This spec currently ties the implementation quite rigidly, we want to assert that the relevant messages are sent something like

```
it "sends a complete cycle of notifications" do
  formatter = double("formatter")
  %w[start seed start_dump dump_pending dump_failures dump_summary close].map(&:to_sym).each do |notification|
    reporter.register_listener formatter, message
    expect(formatter).to receive(message).ordered
  end
  reporter.exit_early(42)
end
```

Not totally certain on the ordering

Reply:
Thanks for the help with the wording. I changed the spec to be like you wanted, although it does not test anything specific to this method and it also passes if I instead call `reporter.report { 42 }`... but whatever.

Reply:
Yes thats the point, you don't have to implement this with `reporter.report { 42 }`, you could also implement it by calling the required notifications, by calling start then finish, etc etc

Reply:
The behaviour of this method is it causes an exit report, and doesn't take a block

Reply:
Yeah, sometimes it's hard to put the boundary between "too attached to the implementation" and "not testing enough". I used the `#report` specs as a base for my initial version, but then I noticed that all unit tests would still pass if I changed the `report` method to just be `yield self`, so I decided to add a more specific assertion.

Thanks for your review, it was tough to match our visions, but I'm glad I could make this acceptable for you!

Reply:
Thanks for your patience to see this over the line